src/python/clawutil/data.py

```diff
@@ -14,6 +14,8 @@
 from __future__ import absolute_import
 from __future__ import print_function
 import os
+from inspect import signature
+
```

Comment:
Is this used?

Reply:
Yeah it's used to pull the default `out_file` for all of the data object `write` methods so that you can set a common `out_dir` and then write all .data files in that directory using their default names. On the other hand, clawpack actually needs specific names for the file, right? So we might not actually *want* to allow for a different `out_file` name. In that case, we could hard-code all the outfile names, but give each write method an `out_dir` kwarg. In that case, we could just pass `out_dir` from `ClawRunData.write` to all the individual data objects `.write()` methods, and in that case we could drop signature.

Reply:
Sorry about that, missed it.

For the other point the code has the capability to use different file names but it is virtually unexposed.  Using introspection may not be wise to do the file names in case someone subclasses a data object and it has a different name.  Right now the file name is defaulted to the correct one.  The containing path though is a good idea.

Reply:
If someone subclasses the data object, wouldn't they pretty much need to create a new write method if they want it to be callable by `ClawRunData.write` and if they want it to not overwrite the superclass's `.data` file?

I noticed that some data objects had an `out_file` kwarg and some had the filename hardcoded, so I basically just switched all of them to having an `out_file`. We could keep the filename and the containing directory separate within the function signature (i.e. having both `out_dir` and `out_file`, where `out_file` is just a file name, rather than letting it be a path). Is that what you're thinking?

That would essentially have the same functionality as using `signature` and prepending `out_dir` to the default `out_file`, no?

Reply:
Yup, that works.  I forgot that we were slightly inconsistent with that.